src/mame/drivers/p2000t.cpp

```diff
@@ -60,6 +135,29 @@ void p2000t_state::p2000t_mem(address_map &map)
 	map(0xe000, 0xffff).bankrw(m_bank);
 }
 
+void p2000h_state::p2000h_mem(address_map &map)
+{
+	map(0x0000, 0xffff).rw(FUNC(p2000h_state::memory_read), FUNC(p2000h_state::memory_write));
+}
+
+u8 p2000h_state::memory_read(offs_t offset)
+{
+	/* if port c0-cf bit 0 is set mem page 0 (0x0000-0x1fff) and 1 (0x2000-0x3fff) is addressed as rom on reading else as RAM */
+	if ((offset < 0x2000) && m_hiresmem_bank0_ROM)
+	{
+		if (m_hiresrom == NULL) {
+			m_hiresrom = memregion("hirescpu")->base();
+		}
+		return m_hiresrom[offset];
+	}
```

Comment:
You’ve mixed K&R and Allman styles here.  Please use consistent code formatting style in a source file.  Also, this isn’t the kind of place you should be doing a tag lookup.